metadoc-js/src/main/scala/metadoc/ScalaDefinitionProvider.scala

```diff
@@ -0,0 +1,45 @@
+package metadoc
+
+import scala.scalajs.js
+import scala.scalajs.js.annotation._
+import org.scalajs.dom
+import scala.meta._
+import metadoc.schema.{Index, Symbol}
+
+@ScalaJSDefined
+class ScalaDefinitionProvider(attrs: Attributes, index: Index) extends MonacoLanguages.DefinitionProvider {
+  override def provideDefinition(
+    model: MonacoEditor.IReadOnlyModel,
+    position: Monaco.Position,
+    token: Monaco.CancellationToken): MonacoLanguages.Definition = {
+
+    val offset = model.getOffsetAt(position)
+
+    val symbol: Option[String] = attrs.names.collectFirst {
+      case (pos, sym) if pos.start.offset <= offset && offset <= pos.end.offset =>
+        sym.syntax
+    }
+    val maybeDefinition = symbol.flatMap { sym =>
```

Comment:
Out of curiosity, would it be possible to do a fetch here? The api seems to support returning a `Thenable<Definition>` https://microsoft.github.io/monaco-editor/api/interfaces/monaco.thenable.html